src/Runner.Listener/MessageListener.cs

```diff
@@ -516,14 +537,11 @@ private bool IsSessionCreationExceptionRetriable(Exception ex)
             }
         }
 
-        private async Task<VssCredentials> GetNewOAuthAuthorizationSetting(CancellationToken token)
+        private async Task<VssCredentials> GetNewOAuthAuthorizationSetting(CancellationToken token, bool adhoc = false)
         {
             Trace.Info("Start checking oauth authorization url update.");
             while (true)
             {
-                var backoff = BackoffTimerHelper.GetRandomBackoff(TimeSpan.FromMinutes(30), TimeSpan.FromMinutes(45));
```

Comment:
move the sleep to the end of the iteration, since most runner has finished migration. there are a small number of runners connected to the server once a while only for a few minutes, so the delay is too long for them.